drivers/usb/udc/udc_smartbond.c

```diff
@@ -0,0 +1,1749 @@
+{
+	return udc_lock_internal(dev, K_FOREVER);
+}
+
+static int udc_smartbond_unlock(const struct device *dev)
+{
+	return udc_unlock_internal(dev);
+}
+
+static const struct udc_api udc_smartbond_api = {
+	.lock = udc_smartbond_lock,
+	.unlock = udc_smartbond_unlock,
+	.device_speed = udc_smartbond_device_speed,
+	.init = udc_smartbond_init,
+	.enable = udc_smartbond_enable,
+	.disable = udc_smartbond_disable,
+	.shutdown = udc_smartbond_shutdown,
+	.set_address = udc_smartbond_set_address,
+	.host_wakeup = udc_smartbond_host_wakeup,
+	.ep_enable = udc_smartbond_ep_enable,
+	.ep_disable = udc_smartbond_ep_disable,
+	.ep_set_halt = udc_smartbond_ep_set_halt,
+	.ep_clear_halt = udc_smartbond_ep_clear_halt,
+	.ep_enqueue = udc_smartbond_ep_enqueue,
+	.ep_dequeue = udc_smartbond_ep_dequeue,
+};
+
+#define DT_DRV_COMPAT renesas_smartbond_usbd
+
+#define UDC_IRQ(inst)      DT_INST_IRQ_BY_IDX(inst, 0, irq)
+#define UDC_IRQ_PRI(inst)  DT_INST_IRQ_BY_IDX(inst, 0, priority)
+#define VBUS_IRQ(inst)     DT_INST_IRQ_BY_IDX(inst, 1, irq)
+#define VBUS_IRQ_PRI(inst) DT_INST_IRQ_BY_IDX(inst, 1, priority)
+
+/*
+ * Minimal transfer size needed to use DMA. For short transfers
+ * it may be simpler to just fill hardware FIFO with data instead
+ * of programming DMA registers.
+ */
+#define DMA_MIN_TRANSFER_SIZE(inst) DT_INST_PROP(inst, dma_min_transfer_size)
+#define FIFO_READ_THRESHOLD(inst)   DT_INST_PROP(inst, fifo_read_threshold)
+
+#define UDC_SMARTBOND_DEVICE_DEFINE(n)                                                             \
+                                                                                                   \
+	static const struct udc_smartbond_config udc_smartbond_cfg_##n = {                         \
+		.udc_irq = UDC_IRQ(n),                                                             \
+		.vbus_irq = VBUS_IRQ(n),                                                           \
+		.dma_min_transfer_size = DMA_MIN_TRANSFER_SIZE(n),                                 \
+		.fifo_read_threshold = FIFO_READ_THRESHOLD(n),                                     \
+		.fifo_read_threshold = FIFO_READ_THRESHOLD(n),                                     \
+		.num_of_eps = DT_INST_PROP(n, num_bidir_endpoints),                                \
+		.dma_cfg =                                                                         \
+			{                                                                          \
+				.tx_chan = DT_INST_DMAS_CELL_BY_NAME(n, tx, channel),              \
+				.tx_slot_mux = DT_INST_DMAS_CELL_BY_NAME(n, tx, config),           \
+				.tx_dev = DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, tx)),         \
+				.rx_chan = DT_INST_DMAS_CELL_BY_NAME(n, rx, channel),              \
+				.rx_slot_mux = DT_INST_DMAS_CELL_BY_NAME(n, rx, config),           \
+				.rx_dev = DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, rx)),         \
+			},                                                                         \
```

Comment:
```c
		.dma_cfg = {                                                                          \
			.tx_chan = DT_INST_DMAS_CELL_BY_NAME(n, tx, channel),              \
			.tx_slot_mux = DT_INST_DMAS_CELL_BY_NAME(n, tx, config),           \
			.tx_dev = DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, tx)),         \
			.rx_chan = DT_INST_DMAS_CELL_BY_NAME(n, rx, channel),              \
			.rx_slot_mux = DT_INST_DMAS_CELL_BY_NAME(n, rx, config),           \
			.rx_dev = DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(n, rx)),         \
		}, 
```

Reply:
This strange formatting was done by `clang-format` tool. I wander if you have power to make someone change `.clang-format` to make it more consistent for nested structure initialization.

Fixed anyway.